packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -544,10 +574,30 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
   /// To close the drawer once it is open, use [Navigator.pop].
   ///
   /// See [Scaffold.of] for information about how to obtain the [ScaffoldState].
-  void openDrawer() {
+  void opendrawer() {
     _drawerKey.currentState?.open();
   }
 
+  void openEndSideDrawer() {
+    _endSideDrawerKey.currentState?.open();
+  }
+
+  bool isDrawerOpen() {
+    if (_drawerKey.currentState?.isOpen() == true) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  bool isEndSideDrawerOpen() {
+    if (_endSideDrawerKey.currentState?.isOpen() == true) {
+      return true;
+    } else {
+      return false;
+    }
+  }
```

Comment:
It looks like you're using these functions to couple the behavior of the start and end drawers.  The way you've done this introduces too tight a coupling.  I'm not clear on what behavior you're trying to create.  Can you explain what end-user visible behavior you're trying to achieve?

Reply:
The idea was to have only one of the drawers open at a time. Otherwise the drawers will overlap on top of one another.

Reply:
That makes sense, but there are a bunch of different interaction designs that might achieve that goal.  For example, you could have an edge-trigger on each drawer's animation leaving the `isDismissed` state that triggers the other drawer to `close()` (and cancels any current gestures interacting with that drawer).

Reply:
Sure. I'll check up on this. Thanks a lot for the feedback.

Reply:
Fixed